### PR TITLE
Warn and exit mach if it detects MinGW Python

### LINKED ISSUE #13644: mach should fail and warn if it detects mingw64 python
If you fail to follow the instructions and rename `/mingw64/bin/python2.7.exe` and `/mingw64/bin/python2.exe` (or if you upgrade mingw64 and they reappear) then mysterious errors will occur.

We should change the `mach` script and `mach_bootstrap.py` so that they detect this case and fail and print an error. The logic will be slightly different in each case since in the `mach` script we'll already be executing under the bad python presumably, so we'll need to check the path to python2 or python2.7.exe is not under `/mingw64/bin`.

### PR DESCRIPTION
**Pull request by UK992**
r? @Wafflespeanut 

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #13644

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13661)

### CODE REVIEW

File: python/mach_bootstrap.py [hunk 1]
```diff
@@ -195,6 +195,14 @@ def bootstrap(topdir):
         print('Current path:', topdir)
         sys.exit(1)
 
+    # We don't support MinGW Python
+    if os.path.join(os.sep, 'mingw64', 'bin') in sys.executable:
+        print('Cannot run mach with MinGW Python.')
+        print('\nPlease execute following commands:')
+        print(' mv /mingw64/bin/python2.exe /mingw64/bin/python2-mingw64.exe')
+        print(' mv /mingw64/bin/python2.7.exe /mingw64/bin/python2.7-mingw64.exe')
```
Comment (Wafflespeanut) [inline python/mach_bootstrap.py]: Nit: We don't have to ask them to follow a specific command. The message could be something like,

```
Please rename /mingw64/bin/python2.exe to /mingw64/bin/python2-mingw64.exe and /mingw64/bin/python2.7.exe to /mingw64/bin/python2.7-mingw64.exe
```

Comment (UK992) [inline python/mach_bootstrap.py]:
```
Cannot run mach with MinGW Python.

Please rename following files:
 /mingw64/bin/python2.exe   -> /mingw64/bin/python2-mingw64.exe
 /mingw64/bin/python2.7.exe -> /mingw64/bin/python2.7-mingw64.exe
```
is this better?

Comment (Wafflespeanut) [inline python/mach_bootstrap.py]: WFM :)

**Comment by Wafflespeanut** [comment]
Thanks!

@bors-servo r+

**Pull request merged**